docs: ng-conf 2020 added

owenmecham opened the pull request:
Added ng-conf 2020 marketing info to events and announcements

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [x] Other... Please describe: Marketing section of angular.io


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

kapunahelewong committed:
docs: edit copy of getting started step 2 (#32776)

PR Close #32776

kapunahelewong committed:
docs: edit copy of getting started step 3 (#32820)

PR Close #32820

dannyskoog committed:
style: enforce disallowance of object constructor (#33211)

Applying the `prefer-literal` tslint rule to object enforces the style guide rule https://google.github.io/styleguide/jsguide.html#features-objects-ctor

PR Close #33211

kapunahelewong committed:
docs: fix migrations formatting (#33267)

PR Close #33267

alxhub committed:
fix(ivy): throw a better error when DI can't inject a ctor param (#33739)

Occasionally a factory function needs to be generated for an "invalid"
constructor (one with parameters types which aren't injectable). Typically
this happens in JIT mode where understanding of parameters cannot be done in
the same "up-front" way that the AOT compiler can.

This commit changes the JIT compiler to generate a new `invalidFactoryDep`
call for each invalid parameter. This instruction will error at runtime if
called, indicating both the index of the invalid parameter as well as (via
the stack trace) the factory function which was generated for the type being
constructed.

Fixes #33637

PR Close #33739

atscott committed:
fix(ivy): align TestBed.overrideProvider with what happens with providers in TestBed providers array (#33769)

In Ivy, if you do:
`TestBed.configureTestingModule({providers: [{provide: Service}]});`
the injector will attempt to inject Service as if it was simply listed
in the providers array like `{providers: [Service]}`
This fixes an inconsistency when similarly providing an override with no
`useValue` or `useFactory`.

PR Close #33769

Commit:
fix(language-service): Insert parentheses for method completion (#33860)

This commit leverages the `insertText` field in `ts.CompletionEntry` to
return a completion text for class methods that includes parentheses.

PR closes https://github.com/angular/vscode-ng-language-service/issues/15

PR Close #33860

Commit:
feat: update the locale files (#33556)

PR Close #33556

pkozlowski-opensource committed:
perf: add js-web-frameworks benchmark (#34034)

PR Close #34034

Commit:
feat: add direction property to locale files (#33556)

PR Close #33556

IgorMinar committed:
build: consolidate @angular-devkit/build-angular to the root package.json (#34002)

This allows us to update the version of the package in a single place for all tests.

Notable exemption of this is aio which currently doesn't depend on anything installed in the root.

PR Close #34002

IgorMinar committed:
build: remove obsolete build related tools and helper scripts (#34058)

none of these files are needed any more as they were replaced by Bazel.

PR Close #34058

ahasall committed:
docs: remove information about which version of Firefox started supporting custom elements (#34007)
PR Close #34007

arturovt committed:
docs(compiler): fix typos and spelling (#34008)

PR Close #34008

JoostK committed:
fix(ngcc): report diagnostics from migrations (#34014)

When ngcc is analyzing synthetically inserted decorators from a
migration, it is typically not expected that any diagnostics are
produced. In the situation where a diagnostic is produced, however, the
diagnostic would not be reported at all. This commit ensures that
diagnostics in migrations are reported.

PR Close #34014

JoostK committed:
fix(ngcc): log Angular error codes correctly (#34014)

Replaces the "TS-99" sequence with just "NG", so that error codes are
logged correctly.

PR Close #34014

JoostK committed:
fix(ngcc): fix undecorated child migration when `exportAs` is present (#34014)

The undecorated child migration creates a synthetic decorator, which
contained `"exportAs": ["exportName"]` as obtained from the metadata of
the parent class. This is a problem, as `exportAs` needs to specified
as a comma-separated string instead of an array. This commit fixes the
bug by transforming the array of export names back to a comma-separated
string.

PR Close #34014

clydin committed:
docs: Ivy i18n guide updates for localize option (#34053)

PR Close #34053

sonukapoor committed:
test(docs-infra): ensure RxJS-related docs examples can compile without errors (#34063)

Previously, some RxJS-related examples (which are not proper Angular apps) were not
tested on CI as part of the `example-e2e` npm script. This meant that the examples
could get out-of-date or contain compile errors without as noticing.

This commit ensures that the `example-e2e` script picks up these examples and checks
that they compile successfully.

Partly addresses #28017.

PR Close #34063

DenysVuika committed:
docs: update Angular books list (#34062)

PR Close #34062

gregmagolan committed:
build: update to nodejs rules 0.42.1 (#34073)

* This brings in a fix to the `@npm//foo:foo_files` targets for https://github.com/angular/angular/pull/33927 so the a rules_nodejs patch can be removed.
* It also brings a protractor_web_test fix that resolves the need for a work-around in /modules/playground/e2e_test/sourcemap/BUILD.bazel.

PR Close #34073

ajitsinghkaler committed:
docs: add deprecated-api-item class to deprecated items (#34088)

Fixes #31455

PR Close #34088

ayazhafiz committed:
fix(language-service): determine correct type for ngFor exported values (#34089)

Currently, variables of an unknown type in an `*ngFor` expression are
refined to have the type of the iterable binding of the `*ngFor`
expression. Unfortunately, this is a bug for variables aliasing
[values exported by
`*ngFor`](https://angular.io/api/common/NgForOf#local-variables),
including `index` and `first`, because they are also given the type of
the binding expression, but they are not of the binding type. For
example, in

```typescript
@Component({
  selector: 'test',
  template: `
    <div *ngFor="let hero of heroes; let i = index; let isFirst = first">
      {{ hero }}
    </div>
  `
})
export class TestComponent {
  heroes: Hero[];
}
```

The local variables `i` and `isFirst` are determined to have a type of
`Hero`, when actually their types are `number` and `boolean`,
respectively.

This commit fixes this bug by checking if the value of a variable in an
`*ngFor` expression is known to be an export and assigning the variable
the type of that export value. Only if the variable does not alias an
export is it typed with the binding value of the `*ngFor` expression.

Closes https://github.com/angular/vscode-ng-language-service/issues/460

PR Close #34089

josephperrott committed:
build: Add deprecation warning when compile=aot is used for building with Ivy (#34109)

To inform downstream users to switch to using angular_ivy_enabled as the build
variable for setting Ivy, a deprecation message is printed instructing the user
to migrate away from building with compile=*

PR Close #34109

josephperrott committed:
build: add failure message if compile build variable for angular/angular usage (#34109)

To quicken migration for our own developers away from using compile=aot
for setting ivy, we actually fail the build process if the compile
build variable is used with a message to use our config flags instead.

PR Close #34109

Commit:
fix(language-service): Do not produce diagnostics if metadata for NgModule not found (#34113)

The language service incorrectly reports an error if it fails to find
NgModule metadata for a particular Component / Directive. In many cases,
the use case is legit, particularly in test.

This commit removes such diagnostic message and cleans up the interface
for `TypeScriptHost.getTemplateAst()`.

PR closes https://github.com/angular/vscode-ng-language-service/issues/463

PR Close #34113

kara committed:
refactor(ivy): remove unnecessary fac wrapper (#34076)

For injectables, we currently generate a factory function in the
injectable def (prov) that delegates to the factory function in
the factory def (fac). It looks something like this:

```
factory: function(t) { return Svc.fac(t); }
```

The extra wrapper function is unnecessary since the args for
the factory functions are the same. This commit changes the
compiler to generate this instead:

```
factory: Svc.fac
```

Because we are generating less code for each injectable, we
should see some modest code size savings. AIO's main bundle
is about 1 KB smaller.

PR Close #34076

Commit:
fix(language-service): use host.error() instead of console.error() (#34114)

`host.error()` would log to file, and makes error messages much easier
to inspect because entries are time-stamped.

PR Close #34114

kara committed:
refactor(ivy): do not generate providedIn: null (#34116)

We should only generate the `providedIn` property in injectable
defs if it has a non-null value. `null` does not communicate
any information to the runtime that isn't communicated already
by the absence of the property.

This should give us some modest code size savings.

PR Close #34116

Commit:
fix(language-service): Make missing module suggestion instead of error (#34115)

If a Component or Directive is not part of any NgModule, the language
service currently produces an error message. This should not be an
error. Instead, it should be a suggestion.

This PR removes `ng.DiagnosticKind`, and instead reuses
`ts.DiagnosticCategory`.

PR closes https://github.com/angular/vscode-ng-language-service/issues/458

PR Close #34115

devversion committed:
docs: update title of v9 injectable migration guide (#34125)

The missing-injectable migration has been updated to handle a breaking change that is
unrelated to missing ´@Injectable` decorators. Though, the breaking change will be handled
as part of this migration since we did not want to create another migration (with all the boilerplate etc.)

The guide has been already updated to reflect the new pattern the migration handles, but we
should also rename the title of the guide to something that also mentions the other pattern.

Not renaming the guide URL since it is referenced in past releases and it's safer to keep the old
URL. The important thing is to change the actual rendered title.

PR Close #34125

devversion committed:
refactor(core): update missing-injectable migration description to match guide (#34125)

Updates the description of the `missing-injectable` migration to mention the second pattern
that will be automatically handled.

PR Close #34125

Zuckjet committed:
docs(core): update render factory description (#34137)

PR Close #34137

crisbeto committed:
fix(common): reflect input type in NgForOf context (#33997)

Fixes `NgForOf` not reflecting the type of its input in the `NgForOfContext`.

PR Close #33997

santoshyadavdev committed:
docs: small typo in DI docs (#34122)

PR Close #34122

IgorMinar committed:
build: remove gulp-connect and cors npm dependencies (#34058)

we no longer need them.

PR Close #34058

IgorMinar committed:
build: remove fsevents from package.json (#34058)

we no longer need it.

PR Close #34058

IgorMinar committed:
refactor(compiler-cli): remove bogus packages/compiler-cli/integrationtest/benchmarks/ (#34142)

nobody uses this folder for anything and the README.md is misleading.

PR Close #34142

IgorMinar committed:
build: support yarn 1.19.x (#34143)

PR Close #34143

IgorMinar committed:
fix(bazel): reenable template type checking in ng_module (#34144)

due to an unfortunate condition in https://github.com/angular/angular/blob/168abc6d6f52713383411b14980e104c99bfeef5/packages/compiler-cli/src/ngtsc/program.ts#L430-L434 the typechecking has been disabled when running under bazel + ivy.

As far as I can tell the ivyTemplateTypeCheck flag is now obsolete, so removing this
code from ng_module.bzl is desirable. I'll send a separate PR to remove the flag completely.

PR Close #34144

IgorMinar committed:
build: re-enable template type-checking for various targets across the repo (#34144)

Various targets have their template type-checking disabled in the past.

There is no reason for this any more.

The only target that was tricky was packages/examples/core:core_examples
which was quite broken and I had to fix it up.

Template typechecking is still disabled under blaze, see FW-1753 for more
info.

PR Close #34144

IgorMinar committed:
ci: replace fixme-ivy-aot with no-ivy-aot tag for //packages/language-service/test (#34144)

This target will need to be completely updated once the language service supports Ivy.

PR Close #34144

petebacondarwin committed:
fix(ivy): i18n - correctly parse XLIFF placeholders (#34155)

The ViewEngine translation extractor does not convert `-` to `_` for
placeholders that represent custom elements. For example `<app-component>`
gets converted to placeholders like `START_TAG_APP-COMPONENT`.

In `$localize` placeholders are expected to be snake-case, not kebab-case.
So we must normalize them when parsing a translation file that might have
been created via the View Engine translation extractor.

The `$localize` extraction code will normalize these placeholders when
creating translation files in the first place.

Fixes #34151

PR Close #34155

pkozlowski-opensource committed:
fix(ivy): allow insertion of views attached to a different container (#34156)

Fixes #34152

PR Close #34156

pkozlowski-opensource committed:
fix(ivy): consistenly return -1 from ViewContainerRef.indexOf for non-inserted view (#34156)

PR Close #34156

crisbeto committed:
fix(common): reflect input type in NgIf context (#33997)

Fixes the content of `NgIf` being typed to any.

Fixes #31556.

PR Close #33997

JiaLiPassion committed:
fix: implement Symbol.specics of Promise (#34162)

Close #34105, #33989

PR Close #34162

alan-agius4 committed:
docs: remove redundant dashes (#34172)
PR Close #34172

ivanwonder committed:
fix(language-service): bug of accessing a string index signature using dot notation (#34177)

PR Close #34177

owenmecham committed:
docs: ng-conf 2020 added

Added ng-conf 2020 marketing info to events and announcements

owenmecham committed:
docs: ng-conf 2020 added

Added ng-conf 2020 marketing info to events and announcements

owenmecham committed:
docs: ng-conf 2020 added

Added ng-conf 2020 marketing info to events and announcements

ivanwonder committed:
fix(language-service): return the js primitive type name (#34177)

PR Close #34177

gkalpak commented on aio/content/marketing/announcements.json:
```diff
@@ -1,15 +1,15 @@
-    "startDate": "2019-04-02",
-    "endDate": "2019-04-20",
-    "message": "Join us for ng-conf<br/>May 1st-3rd, 2019",
+    "startDate": "2020-01-01",
```
Did you mean "2019-**03**-01"? We usually show the bnotification for about a month. Three months might be too long.
(cc @StephenFluin)

owenmecham replied:
I will make an adjustment and then push my changes given those guidelines.

gkalpak reviewed (changes requested):
Thx! Left a couple of minor comments.
(BTW, are you involved with the conf organization. If not, we should probably check with the organizing team to make sure thay are happy with the content.)

pkozlowski-opensource committed:
test(ivy): ViewContainerRef.move where the old and new indexes are the same (#34156)

PR Close #34156

alan-agius4 committed:
fix(bazel): don't rely on @angular/core being as a depedency to install @angular/bazel (#34181)

With this change we fix the logic to detect if a package is installed, removing a package and add a package by using the CLI schematic helpers.

Also we save `@angular/bazel` package directly as a `devDependency` when doing `ng-add`.

Closes #34164

PR Close #34181

crisbeto committed:
perf(ivy): avoid duplicate state lookup and default function parameters (#34183)

Includes a few minor performance improvements:
* In the `nextContext` instruction we assign a new LView to the `LFrame.contextLView` and then we immediately look it up to get its context. We don't need to do that since we know the view that was assigned already.
* Removes the default value for the `level` parameter of `nextContextImpl` because it generates more code in es5 and is internal-only.
* Removes the default parameter from `setActiveHostElement` since it generates extra code and it's an internal function.
* Makes a check in `setElementExitFn` more strict since we're guaranteed for the value to match the type.

PR Close #34183

KwintenP committed:
docs: add kwinten pisman to GDE resources (#34184)

PR Close #34184

owenmecham commented on aio/content/marketing/announcements.json:
```diff
@@ -1,15 +1,15 @@
+    "startDate": "2020-01-01",
+    "endDate": "2020-03-31",
+    "message": "Join us for ng-conf<br/>April 1st-3rd, 2020",
     "imageUrl": "generated/images/marketing/home/ng-conf.png",
```
I can supply a new version of the logo for ng-conf 2020 if you can provide the dimensions that you require.

StephenFluin replied:
You should replace and use the same dimensions as https://github.com/angular/angular/blob/master/aio/content/images/marketing/home/ng-conf.png (the current image) so it just keeps working.

owenmecham committed:
docs: ng-conf 2020 updates

Updated announcement duration and html formatting

owenmecham committed:
docs: ng-conf 2020 updates

Updated announcement duration and html formatting

owenmecham commented:
@gkalpak I am a new member of the ng-conf organizers. Aaron Frost asked me to submit this PR.

jbogarthyde committed:
docs: clarify differential load doc (#34189)

PR Close #34189

crisbeto committed:
fix(ivy): incorrectly validating html foreign objects inside svg (#34178)

Fixes ngtsc incorrectly logging an unknown element diagnostic for HTML elements that are inside an SVG `foreignObject` with the `xhtml` namespace.

Fixes #34171.

PR Close #34178

AndrewKushnir committed:
fix(ivy): support ICUs with pipes (#34198)

Prior to this commit, i18n runtime code failed with the exception saying that no provider was found for ChangeDetectorRef for a pipe used in ICU. The problem happened because the underlying `createViewRef` function was not taking into account IcuContainer as a valid TNodeType. This commit updates the `createViewRef` function to return corresponding ViewRef for TNodeType.IcuContainer.

PR Close #34198

atscott committed:
build: disable buildifier print warning for compile=aot deprecation (#34199)

PR Close #34199

AndrewKushnir committed:
fix(ivy): correctly support `ngProjectAs` on templates (#34200)

Prior to this commit, if a template (for example, generated using structural directive such as *ngIf) contains `ngProjectAs` attribute, it was not included into attributes array in generated code and as a result, these templates were not matched at runtime during content projection. This commit adds the logic to append `ngProjectAs` values into corresponding element's attribute arrays, so content projection works as expected.

PR Close #34200

ivanwonder committed:
fix(language-service): apply suggestion (#34177)

PR Close #34177

AndrewKushnir committed:
refactor(ivy): check metadata presence before compiling Type in R3TestBed (#34204)

Prior to this commit, there was no check in R3TestBed to verify that metadata is resolved using a given Type. That leads to some cryptic error messages (when TestBed tries to compile a Type without having metadata) in case TestBed override functions receive unexpected Types (for example a Directive is used in `TestBed.overrideComponent` call). This commit adds the necessary checks to verify metadata presence before TestBed tries to (re)compile a Type.

PR Close #34204

mprobst committed:
refactor(platform-browser): avoid mutable exports. (#34207)

Previously, browser_util would export a mutable `let` binding that was
initialized as a side-effect of `BrowserDetection.setup()`. This change
refactors the mutable binding into a `const` binding that is immediately
initialized in its initialized.

This is functionally equivalent, but makes it easier for module
optimizers such as Closure Compiler to track down side effects and prune
modules. It is also arguably cleaner to read (no worries about later
changes to the apparently mutable but effectively const binding).

PR Close #34207

petebacondarwin committed:
test: do not store generated i18n files in git (#34135)

PR Close #34135

petebacondarwin committed:
test: do not copy address to the clipboard in integration test (#34135)

PR Close #34135

petebacondarwin committed:
refactor(ivy): i18n - remove hack around TS tagged literals (#34135)

When first written there was no way to specify the raw text when
programmatically creating a template tagged literal AST node.

This is now fixed in TS and so the hack is no longer needed.

PR Close #34135

petebacondarwin committed:
refactor(ivy): i18n - store legacy message ids separately to custom ids (#34135)

This change will enable the Angular compiler to provide these legacy
message ids by default, which will solve problems with ngcc not knowing
whether to generate legacy ids or not.

PR Close #34135

petebacondarwin committed:
refactor(ngcc): store whether to render legacy i18n message ids in the bundle (#34135)

Placing this configuration in to the bundle avoids having to pass the
value around through lots of function calls, but also could enable
support for different behaviour per bundle in the future.

PR Close #34135

petebacondarwin committed:
test(ngcc): tidy up helper function (#34135)

Thanks to @gkalpakl for the better regular expression approach.

PR Close #34135

petebacondarwin committed:
refactor(compiler): i18n - render legacy i18n message ids (#34135)

Now that `@angular/localize` can interpret multiple legacy message ids in the
metablock of a `$localize` tagged template string, this commit adds those
ids to each i18n message extracted from component templates, but only if
the `enableI18nLegacyMessageIdFormat` is not `false`.

PR Close #34135

petebacondarwin committed:
fix(ngcc): render legacy i18n message ids by default (#34135)

By ensuring that legacy i18n message ids are rendered into the templates
of components for packages processed by ngcc, we ensure that these packages
can be used in an application that may provide translations in a legacy
format.

Fixes #34056

PR Close #34135

petebacondarwin committed:
refactor(compiler): remove redundant `metaFromI18nMessage()` function (#34154)

PR Close #34154

petebacondarwin committed:
fix(ivy): i18n - trim whitespace when parsing metadata (#34154)

It is possible for HTML formatters to add whitespace
around the content of `i18n` attribute values. This can
make the meaning and custom ids brittle to simple
whitespace formatting.

This commit ensures that the metadata string extracted
from HTML `i18n` attributes is trimmed before being parsed
into meaning, description and custom id.

PR Close #34154

petebacondarwin committed:
fix(compiler): i18n - trim whitespace from i18n custom ids (#34154)

Fixes #34147

PR Close #34154

petebacondarwin committed:
fix(core): ensure that `ngI18nClosureMode` is guarded (#34211)

If the `ngI18nClosureMode` global check actually makes it
through to the runtime, then checks for its existence should
be guarded to prevent `Reference undefined` errors in strict
mode.

(Normally, it is stripped out by dead code elimination during
build optimization.)

PR Close #34211

crisbeto committed:
refactor(common): add defaults to new generic parameters (#34206)

This is a follow-up to #33997 where some new generic parameters were added without defaults which is technically a breaking change. These changes add the defaults.

PR Close #34206

petebacondarwin committed:
fix(compiler-cli): ensure that `ngI18nClosureMode` is guarded in generated code (#34211)

If the `ngI18nClosureMode` global check actually makes it
through to the runtime, then checks for its existence should
be guarded to prevent `Reference undefined` errors in strict
mode.

(Normally, it is stripped out by dead code elimination during
build optimization.)

This comment ensures that generated template code guards
this global check.

PR Close #34211

petebacondarwin committed:
fix(ngcc): ensure that bundle `rootDir` is the package path (#34212)

Previously the `rootDir` was set to the entry-point path but
this is incorrect if the source files are stored in a directory outside
the entry-point path. This is the case in the latest versions of the
Angular CDK.

Instead the `rootDir` should be the containing package path, which is
guaranteed to include all the source for the entry-point.

---

A symptom of this is an error when ngcc is trying to process the source of
an entry-point format after the entry-point's typings have already been
processed by a previous processing run.

During processing the `_toR3Reference()` function gets called which in turn
makes a call to `ReflectionHost.getDtsDeclaration()`. If the typings files
are also being processed this returns the node from the dts typings files.

But if we have already processed the typings files and are now processing
only an entry-point format without typings, the call to
`ReflectionHost.getDtsDeclaration()` returns `null`.

When this value is `null`, a JS `valueRef` is passed through as the DTS
`typeRef` to the `ReferenceEmitter`. In this case, the `ReferenceEmitter`
fails during `emit()` because no `ReferenceEmitStrategy` is able to provide
an emission:

1) The `LocalIdentifierStrategy` is not able help because in this case
`ImportMode` is `ForceNewImport`.
2) The `LogicalProjectStrategy` cannot find the JS file below the `rootDir`.

The second strategy failure is fixed by this PR.

Fixes https://github.com/angular/ngcc-validation/issues/495

PR Close #34212

petebacondarwin committed:
test(ngcc): update ngcc integration test dependencies (#34212)

This commit updates `@angular/material` and `@angular/cdk` to the
latest release candidate. Doing so exposed a bug in ngcc, which is fixed
by the preceding commit. Also the layout of these libraries changed, so
the checks in the integration test need a bit of tweaking.

PR Close #34212

gkalpak committed:
fix(docs-infra): avoid version mismatch error when using local Angular packages (#34213)

The local Angular packages used to build `aio/` when running
`yarn setup-local`/`yarn build-local` (and related commands), are built
by bazel. Bazel, determines the version to use for these locally built
packages based on the latest tag for a commit on the current branch.
(This works as expected during the release, because the packages are
built on the correct branch with up-to-date tags.)

During local development, however, this often leads to generating older
versions than what the current `@angular/cli` version is compatible
with, if the user has not fetched the latest tags from `angular/angular`
or the branch has not been rebased recently.

Previously, the above (common) situation would result in a build error
by the CLI. (Note that this would be a false error, because in this case
the version set by bazel would not reflect the actual version of the
local packages.) The solution would be for the user to fetch the latest
tags from `angular/angular`, rebase their branch and run a bazel build
again (ensuring that it would actually build anew and not emit cached
artifacts). This was cumbersome and most people didn't even know about
it.

This commit avoids this error and the associated pain-points by
overwriting the versions of the installed local packages with fake
versions based on the ones in the lockfile, which are guaranteed to be
compatible with the currently used CLI version.

Fixes #34208

PR Close #34213

gkalpak committed:
docs(service-worker): fix paths for `ngsw-config` command to match latest CLI (#34215)

PR Close #34215

kapunahelewong commented:
Hey, @owenmecham! Thanks for this. Once you've addressed all the comments, I can help you get the PR through the rest of the process if you like. When you're ready, you'll want to squash and possibly rebase, and I have [notes](https://gist.github.com/kapunahelewong/567c30f7b353fd1c39bbf6d8343e2dca#how-to-rebase) on my own workflow for all of that if you'd like to refer to them. When you're ready or if you have any questions in the meantime, please tag me.

josephperrott committed:
build: add karma-sauce-launcher dependency for karma_web_test macro (#34220)

Currently our bazel saucelabs tests silently fail as it does not have
karma-sauce-launcher available from npm.  By providing it as expected
we will properly run the bazel saucelabs tests once more

PR Close #34220

Commit:
docs: add David Shevitz to about page (#34222)

PR Close #34222

ajitsinghkaler committed:
fix(docs-infra): add deprecated-api-item class to remaining deprecated items (#34192)

Fixes #31455

PR Close #34192

AndrewKushnir committed:
test(ivy): update `ngI18nClosureMode` flag usage in tests (#34224)

Commit that updated i18n message ids rendering (https://github.com/angular/angular/commit/e524322c43ce31c5c6d45bcdc320338b9bd66249) also introduced a couple tests that relied on a previous version of `ngI18nClosureMode` flag format. The `ngI18nClosureMode` usage format was changed in the followup commit (https://github.com/angular/angular/commit/c4ce24647bb45b54db9ad37a21c36e65658fea68) and triggered a problem with the mentioned tests. This commit updates the tests to a new `ngI18nClosureMode` flag usage format.

PR Close #34224

alxhub committed:
fix(ivy): properly parenthesize ternary expressions when emitted (#34221)

Previously, ternary expressions were emitted as:

condExpr ? trueCase : falseCase

However, this causes problems when ternary operations are nested. In
particular, a template expression of the form:

a?.b ? c : d

would have compiled to:

a == null ? null : a.b ? c : d

The ternary operator is right-associative, so that expression is interpreted
as:

a == null ? null : (a.b ? c : d)

when in reality left-associativity is desired in this particular instance:

(a == null ? null : a.b) ? c : d

This commit adds a check in the expression translator to detect such
left-associative usages of ternaries and to enforce such associativity with
parentheses when necessary.

A test is also added for the template type-checking expression translator,
to ensure it correctly produces right-associative expressions for ternaries
in the user's template.

Fixes #34087

PR Close #34221

ivanwonder committed:
refactor(language-service): refactor the method TypeWrapper.name and PipeSymbol.selectSignature (#34177)

PR Close #34177

alan-agius4 committed:
docs: add documentation for server and Ivy without bundleDependencies (#34225)

PR Close #34225

MrJithil committed:
docs(core): add new example for HostListener (#34228)

PR Close #34228

mprobst committed:
refactor(upgrade): avoid mutable exports. (#34232)

Previously, create_angular_testing_module would export a mutable `let`
binding. The binding is already exporting using an accessor function
though, so the export on the let variable seems like an accidental
oversight.

This is functionally equivalent, but makes it easier for module
optimizers such as Closure Compiler to track down side effects and prune
modules.

PR Close #34232

josephperrott committed:
ci: update saucelabs to use angular-framework account (#34233)

Currently all saucelabs usage in our repos is done using the same
account angular-ci.  By migrating to use individual accounts
for each repo, we can better track the usage for each repo as well
as providing concurrency limiting on a per repo basis.

Additionally, we no longer use two separate accounts based on being
on master or a PR branch, so this logic can be removed.

PR Close #34233

alxhub committed:
fix(compiler): switch to modern diagnostic formatting (#34234)

The compiler exports a `formatDiagnostics` function which consumers can use
to print both ts and ng diagnostics. However, this function was previously
using the "old" style TypeScript diagnostics, as opposed to the modern
diagnostic printer which uses terminal colors and prints additional context
information.

This commit updates `formatDiagnostics` to use the modern formatter, plus to
update Ivy's negative error codes to Angular 'NG' errors.

The Angular CLI needs a little more work to use this function for printing
TS diagnostics, but this commit alone should fix Bazel builds as ngc-wrapped
goes through `formatDiagnostics`.

PR Close #34234

Commit:
fix(language-service): Remove getExternalFiles() (#34260)

This commit removes the `getExternalFiles()` from the tsserver plugin.
This API is no longer needed now that we do not intend to support
external templates under the plugin mode.
Instead, the external files are added to the project only when they are
opened by the user.
For complete discussion, see https://github.com/angular/vscode-ng-language-service/issues/473

PR closes https://github.com/angular/vscode-ng-language-service/issues/469
PR closes https://github.com/angular/vscode-ng-language-service/issues/473

PR Close #34260

josephperrott committed:
build: set upper version limit for yarn to <2 (#34236)

Rather than bumping up the allowed version of yarn on each release
we should instead just allow for anything within the major version
1 range.

PR Close #34236

Commit:
build: Enable ivy in google3 if ng_module attr is True (#34238)

This commit creates a way for `ng_module` rule to compile with Ivy based
on the `ivy` attribute. This enables google3 targets to pick the
compiler without using `define` flags.

PR Close #34238

AndrewKushnir committed:
fix(ivy): do not invoke change detection for destroyed views (#34241)

Prior to this commit, calling change detection for destroyed views resulted in errors being thrown in some cases. This commit adds a check to make sure change detection is invoked for non-destroyed views only.

PR Close #34241

josephperrott committed:
fix(common): update closure locales to include directionality data (#34240)

When directionality data was added to @angular/common locales, only
the externally used locales were updated.  We need to additionally
update the closure locales which are synced into google3.

PR Close #34240

gregmagolan committed:
build: update to rules_nodejs 0.42.2 (#34243)

This release brings two bug fixes we're waiting on:
1) fix(builtin): additional_root_paths in pkg_web should also include paths in genfiles and bin dirs (https://github.com/bazelbuild/rules_nodejs/pull/1402), which is a pre-req for https://github.com/angular/angular/pull/34112
2) fix(typescript): fix for cross platform ts_devserver issue #1409 (https://github.com/bazelbuild/rules_nodejs/pull/1413) which resolves ts_devserver launcher template is platform specific #1409 (https://github.com/bazelbuild/rules_nodejs/issues/1409) --- this fixes an OSX developer workflow with --config=remote

This does not upgrade integration/bazel or integation/schematics. That will be done in another PR.

PR Close #34243

pkozlowski-opensource committed:
refactor(ivy): minor cleanup in the listener instruction (#34255)

PR Close #34255

ajitsinghkaler committed:
fix(docs-infra): make header full-width and panels full-width on mobile screens (#34188)

On events page the header was not able to take full width when body exceeds viewport width of the screen So made the below body go overflow-x auto and resources page was taking 80% of the width which is okay on desktop but on mobile it should take 100% width put a media quer for it.

Fixes #34163

PR Close #34188

Commit:
perf: rename index to index_aot (#34258)

The renaming is needed so that js-web-frameworks benchmark test works
inside google.

PR Close #34258

Commit:
fix(language-service): Simplify resolution logic in banner (#34262)

Due to a bug in the existing banner, `typescript` module was require-d
instead of reusing the module passed in from tsserver.
This bug is caused by some source files in language-service that imports
`typescript` instead of `typescript/lib/tsserverlibrary`.
This is not an unsupported use case, it's just that when typescript is
resolved in the banner we have to be very careful about which modules to
"require".
The convoluted logic in the banner makes it very hard to detect
anomalies. This commit cleans it up and removes a lot of unneeded code.

This commit also removes `ts` import in typescript_host.ts and use `tss`
instead to make it less confusing.

PR Close #34262

petebacondarwin committed:
fix(compiler): ensure localized strings are ES5 compatible for JIT mode (#34265)

Previously the JIT evaluated code for ivy localized strings included
backtick tagged template strings, which are not compatible with ES5
in legacy browsers such as IE 11.

Now the generated code is ES5 compatible.

Fixes #34246

PR Close #34265

crisbeto committed:
feat(docs-infra): add the ability to expose globals (#34237)

Adds the ability to expose global symbols in the API docs via the `@globalApi` tag. Also supports optionally setting a namespace which will be added to the name automatically (e.g. `foo` will be renamed to `ng.foo`). Relevant APIs should also be exported through the `global.ts` file which will show up under `core/global`.

PR Close #34237

mattjanssen committed:
docs(forms): fix a punctuation error (#34278)
PR Close #34278

josephperrott committed:
ci: update sha of components repo for components unit test integrations (#34280)

Before updating to remove the compile build variable, we must update
the components unit test integrations to a sha  in the components
repo which no longer relies on the compile build variable.

PR Close #34280

josephperrott committed:
build: remove compile build variable (#34280)

Remove the remaining check for compile build variable in the ng_module
rule.  Now that components is no longer relying on this build variable
we can safely remove it from our repository.

We will leave the remaining check for if the compile build variable is
used within our own repo, with plans to remove it in a couple months.
It is being kept in place to ensure that uncommitted local scripts for
our own development that we have locally, kill error as expected if
they still reference the compile build variable.

PR Close #34280

AndrewKushnir committed:
docs: update CHANGELOG to add v9.0.0-rc.5 release notes (#34282)

PR Close #34282

sonukapoor committed:
docs: Adds NgModule into back-ticks (#34302)
PR Close #34302

owenmecham committed:
docs: ng-conf marketing image

Update ng-conf marketing image to reflect 2020 theme

owenmecham committed:
docs: ng-conf marketing image

Update ng-conf marketing image to reflect 2020 theme

owenmecham committed:
Merge branch 'ng-conf-2020-marketing' of https://github.com/owenmecham/angular into ng-conf-2020-marketing

kapunahelewong commented:
Hi @owenmecham,  a rebase should clear out all those extra commits that aren't part of this PR. You might like [these notes](https://gist.github.com/kapunahelewong/567c30f7b353fd1c39bbf6d8343e2dca#how-to-rebase) on my own workflow. If you need a hand, let me know and I can help you sort it out - I've done this before myself.🙂

owenmecham committed:
Merge branch 'ng-conf-2020-marketing' of https://github.com/owenmecham/angular into ng-conf-2020-marketing

owenmecham commented:
@kapunahelewong I ran through the rebase steps from your process but it appears that this PR is in a bad place now. There are so few changes that it honestly would be faster for me to simply start again on a new branch and abandon this PR and start over.

kapunahelewong commented:
Aw, ok, @owenmecham. Well, if you run into this again, maybe I can try my hand at it and see if I can sort it out from my end. Please feel free to tag me. 🙂
